src/bluesearch/database/article.py

```diff
@@ -73,6 +87,163 @@ def paragraphs(self) -> Iterable[Tuple[str, str]]:
         """
 
 
+class PubmedXMLParser(ArticleParser):
+    """Parser for PubMed XML files using the JATS Journal Publishing DTD.
+
+    Parameters
+    ----------
+    path
+        The path to the XML file from PubMed.
+    """
+
+    def __init__(self, path: Union[str, Path]) -> None:
+        super().__init__()
+        self.content = ElementTree.parse(str(path))
+
+    @property
+    def title(self) -> str:
+        """Get the article title.
+
+        Returns
+        -------
+        str
+            The article title.
+        """
+        titles = self.content.find(".//title-group/article-title")
+        return self.text_content(titles)
+
+    @property
+    def authors(self) -> Generator[str, None, None]:
+        """Get all author names.
+
+        Yields
+        ------
+        str
+            Every author, in the format "Given_Name(s) Surname".
+        """
+        authors = self.content.findall(
+            ".//contrib-group/contrib[@contrib-type='author']"
+        )
+        for author in authors:
+            given_names = self.text_content(author.find("name/given-names"))
+            surname = self.text_content(author.find("name/surname"))
+            if given_names == "" or surname == "":
+                # In rare cases, an author may not have a given name or a surname,
+                # e.g. it could be an organization. We decide to skip those.
+                continue
+            author_str = given_names + " " + surname
+            yield author_str.strip()
+
+    @property
+    def abstract(self) -> Generator[str, None, None]:
+        """Get a sequence of paragraphs in the article abstract.
+
+        Yields
+        ------
+        str
+            The paragraphs of the article abstract.
+        """
+        abstract_pars = self.content.findall(".//abstract//p")
+        for paragraph in abstract_pars:
+            yield self.text_content(paragraph)
+
+    @property
+    def paragraphs(self) -> Generator[Tuple[str, str], None, None]:
+        """Get all paragraphs and titles of sections they are part of.
+
+        Paragraphs can be parts of text body, or figure or table captions.
+
+        Yields
+        ------
+        section : str
+            The section title.
+        text : str
+            The paragraph content.
+        """
+        # Paragraphs of text body
+        abstract_p = self.content.findall(".//abstract/p")
+        caption_p = self.content.findall(".//caption/p")
+        acknowlegd_p = self.content.findall(".//ack/p")
+
+        exclude_list = abstract_p + caption_p + acknowlegd_p
```

Comment:
What are those `_p` paragraphs and why do we want to exclude them?

Reply:
- Abstract is found in the `abstract` property of the class. So it is to avoid duplicates.
- Caption is saved after and keep the section name `Figure Caption` or `Table Caption`. So it is to avoid duplicates.
- Acknowledgements are the acknowledgements of the paper, I don't think we need to keep those paragraphs.

Reply:
Ah I see :) Would it be possible to leave some comments in the code along the same lines? I think this could help to read the code later.

Reply:
Done in a4f7a73